src/world_model/mdp.py

```diff
@@ -0,0 +1,23 @@
+from typing import Any
+from gymnasium import Env
+
+class MDP(Env):
+    """An abstract base class for (fully observed) Markov Decision Processes, extending Env by the ability to reset the environment to a specific state, and to enquire the current state.
+    
+    All observations are full states, so the environment can be reset to an arbitrary state by providing the respective observation.
+    """
+
+    def __init__(self):
```

Comment:
Redundant, plus might cause side-effects (what if an actual env takes arguments in the constructor?)

Reply:
I'm never sure about these things. So you say derived classes can just leave out `__init__`? Assuming this is the message, I have now removed all empty ``__init__``s from all these classes.